src/config.ts

```diff
@@ -37,7 +44,7 @@ function createConfig({
       return { $type: type, ...item };
     },
 
-    (obj) => nodeTypesByName.get(obj.$type),
+    (obj) => obj.$type,
```

Comment:
here. Previously you could return a type object, now you must return a type name. This means that the type must be known to the Schema at runtime via inspection. This should be the case already! However it's technically possible to have orphaned types (e.g. types that aren't in the tree) but still resolve-able because of reasons.